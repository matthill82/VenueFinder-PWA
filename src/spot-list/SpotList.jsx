import React from "react";
import PropTypes from "prop-types";
import SpotItem from "../spot-item/SpotItem";

const SpotList = ({ venues }) => (
  <div className="spot-table">
    <ul className="spot-table__list">
      {venues.map(venue => (
        <SpotItem
          name={venue.name}
          id={venue.id}
          address={venue.location.address}
          distance={venue.location.distance}
        />
      ))}
    </ul>
  </div>
);

SpotList.propTypes = {
  venues: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      location: PropTypes.shape({
        address: PropTypes.string.isRequired,
        city: PropTypes.string.isRequired,
        distance: PropTypes.number.isRequired
      })
    })
  )
};
export default SpotList;
