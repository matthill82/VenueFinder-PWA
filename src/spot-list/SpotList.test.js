import React from "react";
import SpotList from "./SpotList";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

describe("The SpotList component", () => {
  const props = {
    id: 1,
    venues: []
  };

  it("renders correctly", () => {
    const tree = renderer.create(<SpotList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const component = shallow(<SpotList {...props} />);
    expect(component.exists()).toEqual(true);
  });
});
