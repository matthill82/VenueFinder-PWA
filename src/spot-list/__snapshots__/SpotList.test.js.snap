// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SpotList component renders correctly 1`] = `
<div
  className="spot-table"
>
  <ul
    className="spot-table__list"
  />
</div>
`;
